### DISCUSSION TITLE
Integrate Windows Features (Backlog)

### DISCUSSION THREAD

**Discussion by brys0** [wailsapp/wails#1593] [Ideas]
Here's a list of "backlog" features that Windows 10 (?) has that could be implemented:

#### Jumplists (Taskbar context menus)

Implementation: 

https://github.com/microsoft/Windows-classic-samples/tree/main/Samples/Win7Samples/winui/shell/appshellintegration/AutomaticJumpList

What it looks like:
![jumplist](https://user-images.githubusercontent.com/35676267/179638952-60a437a6-9d2c-4966-8150-b174d48b0a73.png)

#### Taskbar overlay icons (Badge)

Implementation:

https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-itaskbarlist3-setoverlayicon

What it looks like:
![overlay_icon](https://user-images.githubusercontent.com/35676267/179639231-1f5f25f5-1adc-4122-9803-cbeed9fd7678.png)

#### Registering custom protocol / URL scheme

Implementation: 

https://www.technothirsty.com/register-custom-url-protocol-using-csharp/

What it looks like: 

Enter this URL in browser and discord will focus. 
![protocol](https://user-images.githubusercontent.com/35676267/179640230-7b1beddb-2700-434e-b29d-5c22ee05a59e.png)

#### Registering and using Toast notification handler

Implementation: 

https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/send-local-toast-desktop-cpp-wrl

What it looks like: 

![toast_incoming_call](https://user-images.githubusercontent.com/35676267/179640533-fcb5eeb6-9e7f-427b-8c93-4401732bbe8d.png)

That's all I can think of for now that is windows platform specific.

**Comment by brys0** [comment]
Please do not feel challenged to do any of these things as they are (rightfully) a pain in the ass. These would be better equipped as plugins rather than Wails by itself.

**Comment by leaanthony** [comment]
Superb list. Thanks for taking the time to compile it. I think we should either add them to the backlog, or add a way of doing them via plugins

**Reply by brys0** [reply]
For sure! Again, I think the Wails default should be just the raw webview2 implementation. Plugins will/should be able to extend the functionality to extend more of the functionality from the OS.

**Comment by airtonix** [comment]
Since this looks just like the "Additional Actions" feature of the [Desktop File Spec](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#extra-actions) and it looks like it's using a custom protocol to allow these extra actions: 

- just lean on the existing OS method for registering custom protocol (on linux, [this is done in the desktop file](https://github.com/wailsapp/wails/discussions/2617#discussioncomment-5739520))
- create protocol entries for each additional action, or action entries
- map protocol actions to cli args for your desktop app


I doubt anything needs to be done here to be honest.
